python/core/auto_generated/geometry/qgsgeometry.sip.in

```diff
@@ -1777,14 +1780,17 @@ of the node is returned.
 .. versionadded:: 3.0
 %End
 
-    QgsGeometry intersection( const QgsGeometry &geometry ) const;
+    QgsGeometry intersection( const QgsGeometry &geometry, double gridSize = -1 ) const;
 %Docstring
 Returns a geometry representing the points shared by this geometry and other.
 
 If the input is a NULL geometry, the output will also be a NULL geometry.
 
 If an error was encountered while creating the result, more information can be retrieved
 by calling :py:func:`~QgsGeometry.lastError` on the returned geometry.
+
+:param geometry: geometry to perform the operation
+:param gridSize: If this optional argument is provided, the inputs are snapped to a grid of the given size, and the result vertices are computed on that same grid. (Requires GEOS-3.9.0 or higher) /since 3.28
```

Comment:
```suggestion
:param gridSize: If this optional argument is provided, the inputs are snapped to a grid of the given size, and the result vertices are computed on that same grid (since 3.28).
```